packages/shared/pkg/keys/hmac_sha256_test.go

```diff
@@ -0,0 +1,73 @@
+package keys
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"testing"
+)
+
+func TestHMACSha256Hashing_ValidHash(t *testing.T) {
+	key := []byte("test-key")
+	hasher := NewHMACSHA256Hashing(key)
+	content := []byte("hello world")
+	expectedHash := "18c4b268f0bbf8471eda56af3e70b1d4613d734dc538b4940b59931c412a1591"
+	actualHash := hasher.Hash(content)
+
+	if actualHash != expectedHash {
+		t.Errorf("expected %s, got %s", expectedHash, actualHash)
+	}
+}
+
+func TestHMACSha256Hashing_EmptyContent(t *testing.T) {
+	key := []byte("test-key")
+	hasher := NewHMACSHA256Hashing(key)
+	content := []byte("")
+	expectedHash := "2711cc23e9ab1b8a9bc0fe991238da92671624a9ebdaf1c1abec06e7e9a14f9b"
+	actualHash := hasher.Hash(content)
+
+	if actualHash != expectedHash {
+		t.Errorf("expected %s, got %s", expectedHash, actualHash)
+	}
+}
+
+func TestHMACSha256Hashing_MismatchedHash(t *testing.T) {
+	key := []byte("test-key")
+	hasher := NewHMACSHA256Hashing(key)
+	content := []byte("hello world")
+	unexpectedHash := "0000000000000000000000000000000000000000000000000000000000000000"
+	actualHash := hasher.Hash(content)
+
+	if actualHash == unexpectedHash {
+		t.Errorf("hash should not match %s", unexpectedHash)
+	}
+}
```

Comment:
this test is kinda unnecessary with `TestHMACSha256Hashing_ValidHash`, isn't it?